manager/apis/app/v1alpha1/fybrikapplication_types.go

```diff
@@ -37,32 +40,34 @@ type DataRequirements struct {
 	// +required
 	Interface InterfaceDetails `json:"interface"`
 
-	// CopyRequrements include the requirements for copying the data
+	// CopyRequrements include the requirements for explicit requests to copy the data
 	// +optional
-	Copy CopyRequirements `json:"copy,omitempty"`
+	FlowParams FlowRequirements `json:"flowParams,omitempty"`
 }
 
-// DataContext indicates data set chosen by the Data Scientist to be used by his application,
-// and includes information about the data format and technologies used by the application
-// to access the data.
+// DataContext indicates data set being processed by the workload
+// and includes information about the data format and technologies used to access the data.
 type DataContext struct {
-	// DataSetID is a unique identifier of the dataset chosen from the data catalog for processing by the data user application.
+	// DataSetID is a unique identifier of the dataset chosen from the data catalog.  For data catalogs that support multiple sub-catalogs, it includes the catalog id and the dataset id.
+	// When writing a new dataset it is the name provided by the user or workload generating it, and should include the sub-catalog name where appropriate.
 	// +required
 	// +kubebuilder:validation:MinLength=1
```

Comment:
could be empty if only for write

Reply:
If it's write and a new dataset, the user puts here the catalogid/filename.  And the in FlowParams the flag IsNewDataSet should be true